core/embed/bootloader/memory.ld

```diff
@@ -23,7 +23,7 @@ ccmram_end = ORIGIN(CCMRAM) + LENGTH(CCMRAM);
 sram_start = ORIGIN(SRAM);
 sram_end = ORIGIN(SRAM) + LENGTH(SRAM);
 
-_codelen = SIZEOF(.flash) + SIZEOF(.data);
+_codelen = SIZEOF(.flash) + SIZEOF(.data) + SIZEOF(.exidx) + SIZEOF(.align);
```

Comment:
Are you sure about `+ SIZEOF(.align)`? I think that does not belong there.

Reply:
The exidx forces me to be a separate section, otherwise linker complains about having both ordered and unordered sections in .flash. But the code itself needs to be aligned to 512 bytes otherwise the image check fails, so i needed to add .align to ensure that. The size of align needs to be included in _codelen to pass image check. Previously alignment was done inside the .flash, so this wasn't needed.

Feels a bit weird, but i don't know if there is another way around this

Reply:
in the end this is handled differently, using `/DISCARD/ ` in the linker script.